types/mithril/index.d.ts

```diff
@@ -138,6 +138,8 @@ declare namespace Mithril {
         background?: boolean;
         /** Milliseconds a request can take before automatically being terminated. */
         timeout?: number;
+        /** The expected type of the response, as a legal value of XMLHttpRequest.responseType. */
+        responseType?: "" | "arraybuffer" | "blob" | "document" | "json" | "text";
```

Comment:
Let's reuse the existing DOM type - we support everything the browser does.

```suggestion
        responseType?: XMLHttpRequestResponseType;
```

Reply:
Thanks for the suggestion 🙂 It's much better your way.